workflow-bot-app/src/main/java/com/symphony/bdk/workflow/engine/executor/message/SendMessageExecutor.java

```diff
@@ -184,16 +184,26 @@ private static String extractContent(ActivityExecutorContext<SendMessage> execut
     if (execution.getActivity().getContent() != null) {
       return execution.getActivity().getContent();
     } else {
-      String template = execution.getActivity().getTemplate();
-      File file = execution.getResourceFile(Path.of(template));
       Map<String, Object> templateVariables = new HashMap<>(execution.getVariables());
       // also bind our utility functions so they can be used inside templates
       templateVariables.put(UtilityFunctionsMapper.WDK_PREFIX, new UtilityFunctionsMapper());
-      return execution.bdk()
-          .messages()
-          .templates()
-          .newTemplateFromFile(file.getPath())
-          .process(templateVariables);
+
+      if (execution.getActivity().getTemplatePath() != null) {
```

Comment:
we can extract "execution.getActivity()" in a variable and use it since this is repeated :)